### PR TITLE
New scenarios; refactored credential management

### PR DESCRIPTION
**Pull request by brheenan**
- Added about:blank, Office.com launcher, Powerpoint, PowerBI
- Changed the way credentials are handled. Now, a class will search for and return the correct credentials for a scenario, and an instance of the class is passed with Run(...) instead of a list of all credentials.

### CODE REVIEW

File: BrowserEfficiencyTest/CredentialManager.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+//--------------------------------------------------------------
+
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace BrowserEfficiencyTest
+{
+    class CredentialManager
```
Comment (TychosNose) [inline BrowserEfficiencyTest/CredentialManager.cs]: Should specify access level per our coding style spec [https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/coding-style.md). For classes in this project we've been specifying an access level of "Internal".

File: BrowserEfficiencyTest/CredentialManager.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    {
+        List<UserInfo> _logins;
```
Comment (TychosNose) [inline BrowserEfficiencyTest/CredentialManager.cs]: Specify the access level here as "private"

File: BrowserEfficiencyTest/Scenarios/AboutBlank.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+//
+//--------------------------------------------------------------
+
+
+using System.Collections.Generic;
+using OpenQA.Selenium.Remote;
+
+namespace BrowserEfficiencyTest
+{
+    class AboutBlank : Scenario
```
Comment (TychosNose) [inline BrowserEfficiencyTest/Scenarios/AboutBlank.cs]: Specify access level to "internal".

File: BrowserEfficiencyTest/Scenarios/PowerBIBrowse.cs [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+//
+//--------------------------------------------------------------
+
+using System.Collections.Generic;
+using OpenQA.Selenium.Remote;
+using OpenQA.Selenium;
+
+namespace BrowserEfficiencyTest
+{
+    class PowerBIBrowse : Scenario
```
Comment (TychosNose) [inline BrowserEfficiencyTest/Scenarios/PowerBIBrowse.cs]: Specify access level to "internal".

File: BrowserEfficiencyTest/Scenarios/OfficeLauncher.cs [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+namespace BrowserEfficiencyTest
+{
+    class OfficeLauncher : Scenario
+    {
+        public OfficeLauncher()
+        {
+            Name = "officeLauncher";
+        }
+
+        public override void Run(RemoteWebDriver d, string browser, CredentialManager credentialManager)
```
Comment (TychosNose) [inline BrowserEfficiencyTest/Scenarios/OfficeLauncher.cs]: Change the name of the RemoteWebDriver variable from "d" to "driver".

File: BrowserEfficiencyTest/Scenarios/OfficePowerpoint.cs [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+{
+    class OfficePowerpoint : Scenario
+    {
+        public OfficePowerpoint()
+        {
+            Name = "powerpoint";
+            Duration = 60;
+        }
+
+        public override void Run(RemoteWebDriver d, string browser, CredentialManager credentialManager)
```
Comment (TychosNose) [inline BrowserEfficiencyTest/Scenarios/OfficePowerpoint.cs]: Change the name of the RemoteWebDriver variable from "d" to "driver".

File: BrowserEfficiencyTest/Scenarios/PowerBIBrowse.cs [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    {
+        public PowerBIBrowse()
+        {
+            Name = "powerBi";
+            Duration = 70;
+        }
+
+        public override void Run(RemoteWebDriver d, string browser, CredentialManager credentialManager)
```
Comment (TychosNose) [inline BrowserEfficiencyTest/Scenarios/PowerBIBrowse.cs]: Change the name of the RemoteWebDriver variable from "d" to "driver".

File: BrowserEfficiencyTest/UserInfo.cs [hunk 8]
```diff
@@ -35,9 +35,9 @@ internal class UserInfo
         public string Domain { get; set; }
 
         [JsonProperty("UserName")]
-        public string UserName { get; set; }
+        public string Username { get; set; }
 
         [JsonProperty("Password")]
-        public string PassWord { get; set; }
+        public string Password { get; set; }
```
Comment (TychosNose) [inline BrowserEfficiencyTest/UserInfo.cs]: Nice catch. I never noticed this before.

Comment (brheenan) [inline BrowserEfficiencyTest/UserInfo.cs]: Good call, done that too

**Comment by TychosNose** [comment]
Looks good. Merging.

**Pull request merged**